bndtools.core/src/bndtools/launch/BndDependencySourceContainer.java

```diff
@@ -92,8 +92,24 @@ public ISourceContainerType getType() {
                             bundleFile = ResourcesPlugin.getWorkspace().getRoot().getFile(bundlePath);
                         }
                         if (bundleFile != null) {
-                            ArchiveSourceContainer tempArchiveCont = new ArchiveSourceContainer(bundleFile, false);
-                            result.add(tempArchiveCont);
+                            // check to see if this archive came from a repo that encodes the source project name
+                            final String sourceProjectName = runbundle.getAttributes().get("sourceProjectName");
+
+                            if (sourceProjectName != null) {
+                                try {
+                                    IProject sourceProject = ResourcesPlugin.getWorkspace().getRoot().getProject(sourceProjectName);
+
+                                    if (sourceProject.exists()) {
+                                        IJavaProject javaSourceProject = JavaCore.create(sourceProject);
+                                        result.add(new JavaProjectSourceContainer(javaSourceProject));
```

Comment:
Agree, using the archive source container is a descent fallback.  although unless the sources are bundled with that archive it wont be too useful in source lookup.  But adding sources to osgi bundles is quite common practice so an archive source container would be of some use.